app/services/katello/pulp3/content_view_version/importable_repositories.rb

```diff
@@ -22,7 +22,7 @@ def generate!
             product = product_for_metadata_repo(repo)
             fail _("Unable to find product '%s' in organization '%s'" % [repo.product.label, @organization.name]) if product.blank?
 
-            root = product.root_repositories.find { |r| r.label == repo.label }
+            root = product.root_repositories.find { |r| r.content_id == repo.content.id }
```

Comment:
```suggestion
            root = product.root_repositories.find do |r| 
                if r.content_id && r.redhat
                   r.content_id == repo.content.id
                else
                   r.label == repo.label
            end
```

Reply:
We need to be able to handle imports where content_id was not provided in the export.

Reply:
Has a typo. Check the patch below